pkg/controller/jobframework/reconciler.go

```diff
@@ -333,8 +347,20 @@ func (r *JobReconciler) stopJob(ctx context.Context, job GenericJob, object clie
 		}
 	}
 
+	var originalSelectors []map[string]string
 	if wl != nil {
-		job.RestoreNodeAffinity(wl.Spec.PodSets)
+		originalSelectors = r.getNodeSelectorsFromPodSets(wl)
```

Comment:
I'm wondering, is there a reason why we need the lookup to workload here, can't the annotations be the single source of through now?